spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/dynamic/SleuthDynamicAspectJTracingInterceptor.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2013-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.sleuth.dynamic;
+
+import brave.ScopedSpan;
+import brave.Tracer;
+import org.aopalliance.intercept.MethodInvocation;
+
+import org.springframework.aop.IntroductionInterceptor;
+import org.springframework.aop.support.AopUtils;
+import org.springframework.cloud.sleuth.util.SpanNameUtil;
+
+/**
+ * Interceptor for tracing {@link MethodInvocation} objects.
+ *
+ * @author Taras Danylchuk
+ * @since 2.2.0
+ */
+public class SleuthDynamicAspectJTracingInterceptor implements IntroductionInterceptor {
+
+	private static final String CLASS_NAME = "Class name";
+
+	private static final String METHOD_NAME = "Method name";
+
+	private static final String SPAN_NAME_DELIMITER = "::";
+
+	private static final String METHOD_PARAMETER_PREFIX = "Method parameter - ";
+
+	private final Tracer tracer;
+
+	private final boolean isTraceParameters;
+
+	public SleuthDynamicAspectJTracingInterceptor(Tracer tracer,
+			boolean isTraceParameters) {
+		this.tracer = tracer;
+		this.isTraceParameters = isTraceParameters;
+	}
+
+	@Override
+	public Object invoke(MethodInvocation invocation) throws Throwable {
+		ScopedSpan span = startNewSpan(invocation);
+		try {
+			return invocation.proceed();
```

Comment:
this interceptor doesn't cover reactive case  
for e.g. see org.springframework.cloud.sleuth.annotation.ReactorSleuthMethodInvocationProcessor   
maybe it is better to make possible to reuse org.springframework.cloud.sleuth.annotation.SleuthMethodInvocationProcessor bean.   
```
Object process(MethodInvocation invocation, NewSpan newSpan,
		ContinueSpan continueSpan) throws Throwable;
```
Make it more generic and transfer metadata from NewSpan and ContinueSpan annotations to some object so other Interceptors could reuse it with their pointcuts

Reply:
agree, refactored logic accordingly
currently i'm reusing existed logic from annotation feature.
As for reactive case i'm thinking to do in separate PR to keep commit and PR simpler. If it's not okay, I'll update current PR with reactive logic.